Index: xorg-server/hw/xfree86/common/xf86pciBus.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86pciBus.c
+++ xorg-server/hw/xfree86/common/xf86pciBus.c
@@ -1311,15 +1311,23 @@ xf86VideoPtrToDriverList(struct pci_devi
         break;
     case 0x1d17:
         switch (dev->device_id) {
-            case 0x3a03:
-            case 0x3a04:
-                driverList[0] = "zx";
-                break;
-            default:
-                driverList[0] = "zx";
-                break;
-            }
-        break;  
+        case 0x3a03:
+        case 0x3a04:
+            driverList[0] = "zx";
+            break;
+        case 0x3d00:
+        case 0x3d02:
+            driverList[0] = "arise";
+            break;
+        case 0x3d01:
+        case 0x3a05:
+            driverList[0] = "cx4";
+            break;
+        default:
+            driverList[0] = "zx";
+            break;
+        }
+        break;
     default:
         break;
     }
