Index: xorg-server/hw/xfree86/common/xf86DGA.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86DGA.c
+++ xorg-server/hw/xfree86/common/xf86DGA.c
@@ -268,7 +268,8 @@ DGACloseScreen(ScreenPtr pScreen)
     DGAScreenPtr pScreenPriv = DGA_GET_SCREEN_PRIV(pScreen);
 
     mieqSetHandler(ET_DGAEvent, NULL);
-    pScreenPriv->pScrn->SetDGAMode(pScreenPriv->pScrn, 0, NULL);
+    if (pScreenPriv->pScrn->SetDGAMode)
+        pScreenPriv->pScrn->SetDGAMode(pScreenPriv->pScrn, 0, NULL);
     FreeMarkedVisuals(pScreen);
 
     pScreen->CloseScreen = pScreenPriv->CloseScreen;
